rust/src/lib.rs

```diff
@@ -184,24 +186,85 @@ pub fn add_to_linker<T: 'static>(
 ) -> Result<EventChannel> {
     let (ctx, events_tx, events_rx) = InstrumentationContext::new(id);
 
-    // Build up a table of function id to name mapping
+    // Parse the wasm to build up a table of function id to name mapping and check version compat
```

Comment:
Maybe it's time to move this to a new function? It could return a struct with the globals, versions and function names or something.

Reply:
Moved to struct `WasmInstrInfo` with accompanying `::new`.